### DISCUSSION TITLE
Request for assistance in setting up logback.xml

### DISCUSSION THREAD

**Discussion by jasonjente** [discord-jda/JDA#2473] [Questions and Help] [answered]
Hello all, 
I have created my bot but my logging doesn't feel very good as I cannot get the timestamp.
I have tried searching on the internet for some examples on setting this up but i couldn't get anything working

Here is my logback.xml:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
    <appender name="FILE" class="ch.qos.logback.core.FileAppender">
        <file>logs/application.log</file> <!-- Specify the path and filename for your log file -->
        <encoder>
            <pattern>%d{HH:mm:ss.SSS}  [%thread] %-5level %logger{36} %msg%n</pattern>
        </encoder>
    </appender>
    <logger name="chapters.configuration" level="INFO" />
    <logger name="chapters.configuration.Foo" level="DEBUG" />
    <root level="info">
        <appender-ref ref="FILE" /> <!-- Add the file appender to the root logger -->
    </root>
</configuration>
```
And the maven dependencies:

```xml
<properties>
  <springboot.version>2.7.11</springboot.version>
  <sl4j.version>2.0.5</sl4j.version>
  <log4j.version>2.20.0</log4j.version>
</properties>
...
<dependencies>
   <dependency>
      <groupId>org.slf4j</groupId>
         <artifactId>slf4j-api</artifactId>
         <version>${sl4j.version}</version>
   </dependency>
   <dependency>
      <groupId>org.slf4j</groupId>
      <artifactId>slf4j-simple</artifactId>
      <version>${sl4j.version}</version>
   </dependency>
   <dependency>
      <groupId>ch.qos.logback</groupId>
      <artifactId>logback-classic</artifactId>
      <version>1.2.3</version>
   </dependency>
</dependencies>
```

And in my codebase I am using the log like this:

```java
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

private static final Logger LOGGER = LoggerFactory.getLogger(FooBar.class);
  LOGGER.info("{} has requested the FooBar command - ENTER.", event.getUser().getName());
```

and the resulting output will be like this:

```
[JDA MainWS-ReadThread] INFO bot.FooBar - SomeUsername has requested the FooBar command - ENTER.
```

Some info about the codebase:
It is in Java 8, using spring-boot, I am running the bot with maven goal spring-boot:run and my full pom.xml can be found here:
https://github.com/jasonjente/lemurios-bot/blob/main/LemuriosBOT/pom.xml

Could you please provide me with an example on how to add the timestamps on the logger? I really cannot understand from the documentation what is to be done and all the examples I have tried have no changes at all on the output.

Thanks in advance!
Jason.

**Comment by MinnDevelopment** [comment]
Remove slf4j-simple, it is a separate implementation of the SLF4J API and conflicts with your logback implementation.

**Reply by jasonjente** [reply]
Hello and thanks for the prompt reply!

Removing the sl4j-simple raised this warning during the startup:

SLF4J: No SLF4J providers were found.
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See https://www.slf4j.org/codes.html#noProviders for further details.
SLF4J: Class path contains SLF4J bindings targeting slf4j-api versions 1.7.x or earlier.

Could the sl4j be a dependency on some other module causing this issue? will it be solved by hunting it down with mvn dependncy:tree and then excluding these dependencies?

I tried this on one spring-boot dependency that had sl4j 1.7.x and I excluded it, restarted and the I got the same error but I got logs from the initialization/setup of the rest endpoints.

**Reply by MinnDevelopment** [reply, marked as answer]
You should also remove your dependency on slf4j-api since it is incompatible with your implementation version.

**Reply by jasonjente** [reply]
That was it, it is fixed! After reading the documentation again I assume that I thought I had to add the sl4j as a dependency and it caused the issue. Anyways, thanks a lot!
